src/System.IO.Pipes.AccessControl/tests/NamedPipeTests/NamedPipeTest.AclExtensions.cs

```diff
@@ -40,12 +40,27 @@ public void GetAccessControl_NamedPipe_BeforeWaitingToConnect()
             Assert.Throws<InvalidOperationException>(() => client.SetAccessControl(new PipeSecurity()));
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "On Desktop we throw an exception based on the internal PipeStream state. In .NET Core, we do not have access to that member since Pipes.AccessControl is its own assembly.")]
+        public void SetAccessControl_NamedPipeClientStream_Broken()
+        {
+            NamedPipeClientStream client = GetAccessControl_NamedPipeClientStream_Broken();
+            Assert.Throws<InvalidOperationException>(() => client.SetAccessControl(new PipeSecurity()));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "On Desktop we throw an exception based on the internal PipeStream state. In .NET Core, we do not have access to that member since Pipes.AccessControl is its own assembly.")]
+        public void SetAccessControl_NamedPipeClientStream_Broken_Desktop()
+        {
+            NamedPipeClientStream client = GetAccessControl_NamedPipeClientStream_Broken();
+            Assert.Throws<IOException>(() => client.SetAccessControl(new PipeSecurity()));
```

Comment:
Today the extension SetAccessControl method calls code that throws IOException if it is in an invalid state. Could we catch that and rethrow and InvalidOperationException to make the code throw the same exception as .NET Framework? Or is IOException too general to catch and might break other cases?